src/Java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntityTreeFarm.java

```diff
@@ -240,34 +240,45 @@ public int getEuDiscountForParallelism() {
 		return 0;
 	}
 
-	public boolean checkMultiblock(final IGregTechTileEntity aBaseMetaTileEntity, final ItemStack aStack) {
-		int xDir = ForgeDirection.getOrientation(aBaseMetaTileEntity.getBackFacing()).offsetX;
-		int zDir = ForgeDirection.getOrientation(aBaseMetaTileEntity.getBackFacing()).offsetZ;
-		int tAmount = 0;
-		if (!aBaseMetaTileEntity.getAirOffset(xDir, 0, zDir)) {
+	@Override
+	public IStructureDefinition<GregtechMetaTileEntityTreeFarm> getStructureDefinition() {
+		if (STRUCTURE_DEFINITION == null) {
+			STRUCTURE_DEFINITION = StructureDefinition.<GregtechMetaTileEntityTreeFarm>builder()
+					.addShape(mName, transpose(new String[][]{
+							{"XXX", "CCC", "CCC"},
+							{"X~X", "C-C", "CCC"},
+							{"XXX", "CCC", "CCC"},
```

Comment:
just curious, why you don't allow to put hatches on the front face, while the tooltip says 'Any Casing'?

Reply:
Good question, reply is a bit more dumb, I copy pasted from the centrifuge and left it as it was.
I'll change it to accept from the front aswell.

Reply:
The centrifuge should also allow on the front, otherwise all mine will break. They face sideways, but are still technically on the front.

Reply:
Pretty sure it was done to avoid this situations
![javaw_dPSmP3yVsK](https://user-images.githubusercontent.com/17811922/142773120-676f9f8c-3e19-425a-8fb4-291d6167e9c3.png)

Reply:
If people want ugly broken textures centrifuges, let them. Doesn’t effect functionality and it still happens if they wallshare.

Reply:
I agree tbh

Reply:
@GlodBlock thoughts?

Reply:
my eyes will bleed 😵